gateway-server/src/main/java/org/apache/knox/gateway/services/topology/impl/DefaultTopologyService.java

```diff
@@ -906,8 +901,8 @@ public void onFileChange(File file) {
     public boolean accept(File file) {
       boolean accept = false;
       if (!file.isDirectory() && file.canRead()) {
-        String fileName = file.getName();
-        if (isDescriptorFile(fileName) || isXmlTypeDescriptorFile(fileName)) {
+        String extension = FilenameUtils.getExtension(file.getName());
+        if (SUPPORTED_EXTENSIONS.contains(extension)) {
```

Comment:
Did you remove the isDescriptorFile(String) method? It didn't appear to be the case.